core/src/main/java/fr/insee/innovation/marathontokubernetes/core/services/converter/MarathonToKubernetesConverter.java

```diff
@@ -56,59 +61,160 @@
+        resourcesRequests.put("cpu",
+                new QuantityBuilder().withNewAmount((appToConvert.getCpus() * 1000) + "m").build());
         containerBuilder.withResources(new ResourceRequirementsBuilder().withRequests(resourcesRequests).build());
-        Container container = containerBuilder.withName(name).withImage(appToConvert.getContainer().getDocker().getImage()).build();
-
-        return new DeploymentBuilder()
-                .withNewMetadata().withName(name).endMetadata()
-                .withNewSpec().withReplicas(appToConvert.getInstances() == null ? 1 : appToConvert.getInstances()).withNewSelector().withMatchLabels(Map.of("app",name)).endSelector()
-                .withNewTemplate().withNewMetadata().withLabels(Map.of("app",name)).endMetadata()
-                .withNewSpec().withContainers(List.of(container)).endSpec().endTemplate()
-                .endSpec().build();
+        Container container = containerBuilder.withName(name)
+                .withImage(appToConvert.getContainer().getDocker().getImage())
+                .withLivenessProbe(getHealthcheck(appToConvert)).build();
+
+        return new DeploymentBuilder().withNewMetadata().withName(name).endMetadata().withNewSpec()
+                .withReplicas(appToConvert.getInstances() == null ? 1 : appToConvert.getInstances()).withNewSelector()
+                .withMatchLabels(Map.of("app", name)).endSelector().withNewTemplate().withNewMetadata()
+                .withLabels(Map.of("app", name)).endMetadata().withNewSpec().withContainers(List.of(container))
+                .endSpec().endTemplate().endSpec().build();
     }
+
+    private Probe getHealthcheck(App appToConvert) {
+        if (appToConvert.getHealthChecks() != null && appToConvert.getHealthChecks().size() > 0) {
+            HealthCheck healthCheck = appToConvert.getHealthChecks().get(0);
+            Probe probe = new Probe();
+            if (healthCheck.getCommand() == null) {
+                String path = healthCheck.getPath();
+                IntOrString port = new IntOrString(healthCheck.getPortIndex());
+                String scheme = healthCheck.getProtocol();
+                HTTPGetAction httpGet = new HTTPGetAction();
+                httpGet.setPath(path);
+                httpGet.setPort(port);
+                httpGet.setScheme(scheme);
+                probe.setHttpGet(httpGet);
+            } else {
+                ExecAction exec = new ExecAction();
+                List<String> commandList = new ArrayList<>();
+                commandList.add(healthCheck.getCommand().getValue());
+                exec.setCommand(commandList);
+                probe.setExec(exec);
+            }
+            probe.setInitialDelaySeconds(healthCheck.getGracePeriodSeconds());
+            probe.setPeriodSeconds(healthCheck.getIntervalSeconds());
+            probe.setTimeoutSeconds(healthCheck.getTimeoutSeconds());
+            return probe;
+        }
+        return null;
+    }
+
+    // {
```

Comment:
I don't think this should be left here. If it's test material then it should be in `src/test/resources`

Reply:
it's deleted :)